tests/bot/utils/test_redis_cache.py

```diff
@@ -42,95 +50,105 @@ async def test_set_get_item(self):
-    #         self.assertEqual(self.redis[key], value)
-    #
-    #     # Test .keys iteration
-    #     for test_case, key in zip(test_cases, self.redis.keys()):
-    #         value = test_case[1]
-    #         self.assertEqual(self.redis[key], value)
-    #
-    # def test_length(self):
-    #     """Test that we can get the correct len() from the RedisDict."""
-    #     self.redis.clear()
-    #     self.redis['one'] = 1
-    #     self.redis['two'] = 2
-    #     self.redis['three'] = 3
-    #     self.assertEqual(len(self.redis), 3)
-    #
-    #     self.redis['four'] = 4
-    #     self.assertEqual(len(self.redis), 4)
-    #
-    # def test_to_dict(self):
-    #     """Test that the .copy method returns a workable dictionary copy."""
-    #     copy = self.redis.copy()
-    #     local_copy = dict(self.redis.items())
-    #     self.assertIs(type(copy), dict)
-    #     self.assertEqual(copy, local_copy)
-    #
-    # def test_clear(self):
-    #     """Test that the .clear method removes the entire hash."""
-    #     self.redis.clear()
-    #     self.redis['teddy'] = "with me"
-    #     self.redis['in my dreams'] = "you have a weird hat"
-    #     self.assertEqual(len(self.redis), 2)
-    #
-    #     self.redis.clear()
-    #     self.assertEqual(len(self.redis), 0)
-    #
-    # def test_pop(self):
-    #     """Test that we can .pop an item from the RedisDict."""
-    #     self.redis.clear()
-    #     self.redis['john'] = 'was afraid'
-    #
-    #     self.assertEqual(self.redis.pop('john'), 'was afraid')
-    #     self.assertEqual(self.redis.pop('pete', 'breakneck'), 'breakneck')
-    #     self.assertEqual(len(self.redis), 0)
-    #
-    # def test_update(self):
-    #     """Test that we can .update the RedisDict with multiple items."""
-    #     self.redis.clear()
-    #     self.redis["reckfried"] = "lona"
-    #     self.redis["bel air"] = "prince"
-    #     self.redis.update({
-    #         "reckfried": "jona",
-    #         "mega": "hungry, though",
-    #     })
-    #
-    #     result = {
-    #         "reckfried": "jona",
-    #         "bel air": "prince",
-    #         "mega": "hungry, though",
-    #     }
-    #     self.assertEqual(self.redis.copy(), result)
+    async def test_set_item_type(self):
+        """Test that .set rejects keys and values that are not strings, ints or floats."""
+        fruits = ["lemon", "melon", "apple"]
+
+        with self.assertRaises(TypeError):
+            await self.redis.set(fruits, "nice")
+
+    async def test_delete_item(self):
+        """Test that .delete allows us to delete stuff from the RedisCache."""
+        # Add an item and verify that it gets added
+        await self.redis.set("internet", "firetruck")
+        self.assertEqual(await self.redis.get("internet"), "firetruck")
+
+        # Delete that item and verify that it gets deleted
+        await self.redis.delete("internet")
+        self.assertIs(await self.redis.get("internet"), None)
```

Comment:
We rely on `.delete` ignoring/working with non-existing keys elsewhere (in `.pop`). I think it's good to test here that deleting a non-existing key is indeed fine. What do you think?

Reply:
Yep, I guess we're testing that already.